src/ai/defaultai.cc

```diff
@@ -1385,17 +1384,15 @@ void DefaultAI::update_all_buildable_fields(const Time& gametime) {
 				delete &bf;
 				buildable_fields.pop_front();
 				continue;
-			} else {  // field is ours but unusable, obviously with builcaps size 0
-				unusable_fields.push_back(bf.coords);
-				delete &bf;
-				buildable_fields.pop_front();
-				continue;
-			}
-
-		} else {  // just rotating
-			buildable_fields.push_back(&bf);
+			}  // field is ours but unusable, obviously with builcaps size 0
+			unusable_fields.push_back(bf.coords);
```

Comment:
wouldn't it make sense to use emplace here as well?

Reply:
emplace vs push only makes a difference when constructing a new object. What push does is to take an existing object and make a copy of it. emplace takes a bunch of arguments which are forwarded to a constructor call. So when allocating a new object, push creates and copies a temporary variable which can be avoided by using emplace. When we have to copy the object anyway then it does not make a difference as both functions would call the copy constructor exactly once. We could change all occurrences of push to emplace anyway, but this would be purely stylistic and I'm not sure if there aren't occasions where using push instead of emplace allows the compiler to add a move optimization.